elf/compat_test.go

```diff
@@ -0,0 +1,35 @@
+// +build linux
+
+// (c) 2018 ShiftLeft GmbH <suchakra@shiftleft.io>
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package elf
+
+import (
+    "path/filepath"
+	"testing"
+)
+
+func TestGetSyscallFnName(t *testing.T) {
+    fnName, err := getSyscallFnNameWithFile("open", filepath.Join("testdata", "prefix_symbols.txt"))
```

Comment:
I would inline the content of `prefix_symbols.txt` and `symbols.txt` in a big string constant here in `compat_test.go` so that we could easily test more without creating many text files in the git repository. Then we can test the empty file test case I mentioned before. To make that test easy to write, we could move the `ioutil.ReadFile()` in `GetSyscallFnName()` and replace `getSyscallFnNameWithFile()` by `getSyscallFnNameWithKallsyms(syscallName string, kallsymsContent string)`

Reply:
I thought about this, but then I thought that testing `getSyscallFnNameWithFile` will be closest as it also tests file opening. But I can do it this way also if not having separate files make more sense.